src/main/java/com/google/devtools/build/lib/syntax/MethodLibrary.java

```diff
@@ -102,6 +105,19 @@ private static Object findExtreme(
     }
   }
 
+  private static Object evalKey(Object obj, Object key, Location loc, Environment env) throws EvalException, InterruptedException {
+    if (key == Runtime.NONE) {
+      return obj;
+    } else if (key instanceof BuiltinCallable) {
+      return ((BuiltinCallable) key).call(new ArrayList<>(Arrays.asList(obj)),
+              new HashMap<>(), new FuncallExpression(Identifier.of(""), ImmutableList.of()), env);
```

Comment:
Use null instead of new HashMap; the parameter is declared "@Nullable".

Reply:
However it's not declared @nullable in `convertStarlarkArgumentsToJavaMethodArguments`. Setting it to null would throw a null pointer exception in `(FuncallExpression.java:511)`

Reply:
That's a bug in convertStarlarkArgumentsToJavaMethodArguments then. For now, use Collections.emptyMap(), which doesn't allocate a new map.